Source/SwiftLintFramework/Rules/Idiomatic/FileNameNoSpaceRule.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+import SourceKittenFramework
+
+private let typeAndExtensionKinds = SwiftDeclarationKind.typeKinds + [.extension, .protocol]
+
+public struct FileNameNoSpaceRule: ConfigurationProviderRule, OptInRule {
+    public var configuration = FileNameNoSpaceConfiguration(
+        severity: .warning,
+        excluded: [],
+        suffixPattern: "\\.*"
+    )
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "file_name_no_space",
+        name: "File Name No Space",
+        description: "File name should not contain any whitespace.",
+        kind: .idiomatic
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        guard let filePath = file.path,
+            case let fileName = filePath.bridge().lastPathComponent,
+            !configuration.excluded.contains(fileName) else {
+            return []
+        }
+
+        let suffixRegex = regex("(?:\(configuration.suffixPattern))\\z")
```

Comment:
This rule is doing more than just checking for whitespace in file names. Why is that?

Reply:
This is intended to just remove the suffix of the filename. By default it would be the extension. But it's configureable so if a user wanted to all a suffix in file names to include whitespace characters, they could. See `testCustomSuffixPattern` test in the PR.

Reply:
Is that something worth supporting? It seems to me like either you allow whitespace in file names or not at all. Seems odd to allow it in some parts of the name but not others, no?

Reply:
Seems fair, I went back and forth on included it or not. I'll remove it.